kaldb/src/main/java/com/slack/kaldb/clusterManager/SnapshotDeletionService.java

```diff
@@ -184,11 +185,11 @@ protected int deleteExpiredSnapshotsWithoutReplicas() {
 
                               // First try to delete the object from S3, then delete from metadata
                               // store. If for some reason the object delete fails, it will leave
-                              // the
-                              // metadata and try again on the next run.
+                              // the metadata and try again on the next run.
                               URI snapshotUri = URI.create(snapshotMetadata.snapshotPath);
                               LOG.info("Starting delete of snapshot {}", snapshotMetadata);
-                              if (s3BlobFs.exists(snapshotUri)) {
+                              if (!SnapshotMetadata.isLive(snapshotMetadata)
```

Comment:
I think it's better to contain the stale snapshot deletion logic in the indexer since it would be hard to keep both  of these pieces of code in sync and not cause a race condition. So for now, I think it's best to move the isLive check into the filter above and have this logic only deal with the non-live snapshots.